templates/block/block--system-branding-block.html.twig

```diff
@@ -0,0 +1,27 @@
+{#
+/**
+ * @file
+ * Theme override for a branding block.
+ *
+ * Each branding element variable (logo, name, slogan) is only available if
+ * enabled in the block configuration.
+ *
+ * Available variables:
+ * - site_logo: Logo for site as defined in Appearance or theme settings.
+ * - site_name: Name for site as defined in Site information settings.
+ * - site_slogan: Slogan for site as defined in Site information settings.
+ */
+#}
+{% if site_logo %}
+  <div id="logo">
+    <a href="{{ path('<front>') }}" title="{{ 'Home'|t }}" rel="home">
```

Comment:
only conditionally output the `<a>` tag if the page being viewed is not the front page, otherwise there would be a self referencial link on the page